app/controllers/concerns/query_routable_controller.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'active_support/concern'
+
+module QueryRoutableController
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :handle_query_routing, only: :index
+  end
+
+  def index; end
+
+  private
+
+  def handle_query_routing
+    return unless @search_options[:affiliate].present? && @search_options[:query].present?
+
+    if routed_query.present? &&
+       !matching_urls_for(routed_query.url).include?(request.referer)
```

Comment:
may want to look at using exclude instead of negating an include. It's easier to read and better practice.

Reply:
I’m trying to avoid extra refactoring of existing code. This file is not new code; it just appears that way because I moved the file. I cleaned up the rubocop issues, but we decided to skip extra refactoring of existing code for the Rails upgrade.